### PR TITLE
Release ITC resource as soon as we have results

### PR DESCRIPTION
**Pull request by rpiaggio**
This fixes a bug where after a few calls to ITC, it became stuck and no further responses were processed.

### CODE REVIEW

File: common/src/main/scala/workers/WorkerClient.scala [hunk 1]
```diff
@@ -70,7 +68,7 @@ class WorkerClient[F[_]: Concurrent: UUIDGen: Logger, R: Pickler] private (
         case FromServer.Complete(mid) if mid === id      =>
           none
         case FromServer.Error(mid, error) if mid === id  =>
-          throw error
```
Comment (rpiaggio) [inline common/src/main/scala/workers/WorkerClient.scala:73]: This is an unrelated fix.

File: explore/src/main/scala/explore/itc/ItcPanelProps.scala [hunk 2]
```diff
@@ -125,10 +125,12 @@ case class ItcPanelProps(
         mode        <- instrumentRow
       yield beforeStart *>
         ItcClient[IO]
-          .request(
-            ItcMessage.GraphQuery(w, ex.time, exposures, constraints, t, mode)
+          .requestSingle(ItcMessage.GraphQuery(w, ex.time, exposures, constraints, t, mode))
```
Comment (rpiaggio) [inline explore/src/main/scala/explore/itc/ItcPanelProps.scala:128]: This is a syntax improvement.

**Review by toddburnside — APPROVED** [review]
Awesome. I think I was running into the bug and couldn't figure out why ITC wasn't responding sometimes. Thanks!

**Review by cquiroz — APPROVED** [review]
This is great, thanks a lot
